File: Linguagem C/Lista001_intro/Ex_008.c
//8) Fazer um algoritmo para resolver uma equação do segundo grau. Seu algoritmo deve verificar se a equação é
//realmente de segundo grau e tratar o valor do discriminante delta adequadamente.
#include <stdio.h>
#include <math.h>

int main() {
    float a, b, c;
    float delta, x1, x2;

    printf("Digite o coeficiente 'a': ");
    scanf("%f", &a);

    printf("Digite o coeficiente 'b': ");
    scanf("%f", &b);

    printf("Digite o coeficiente 'c': ");
    scanf("%f", &c);

    delta = b * b - 4 * a * c;

    if (delta > 0) {
        x1 = (-b + sqrt(delta)) / (2 * a);
        x2 = (-b - sqrt(delta)) / (2 * a);
        printf("As raizes da equacao sao: x1 = %.2f e x2 = %.2f\n", x1, x2);
    } else if (delta == 0) {
        x1 = -b / (2 * a);
        printf("A equacao possui uma raiz real: x = %.2f\n", x1);
    } else {
        float realPart = -b / (2 * a);
        float imaginaryPart = sqrt(-delta) / (2 * a);
        printf("A equacao possui raizes complexas: x1 = %.2f + %.2fi e x2 = %.2f - %.2fi\n", realPart, imaginaryPart, realPart, imaginaryPart);
    }

    return 0;
}

File: Python/001_Lista_intro/Ex_008.py
"""
8) Fazer um algoritmo para resolver uma equação do segundo grau. Seu algoritmo deve verificar se a equação é 
realmente de segundo grau e tratar o valor do discriminante delta adequadamente.
"""
import math

print("Digite os coeficientes da equação quadrática ax^2 + bx + c = 0:")
a = float(input("Digite o valor de 'a': "))
b = float(input("Digite o valor de 'b': "))
c = float(input("Digite o valor de 'c': "))

# Verifica se a equação é realmente de segundo grau
if a == 0:
    print("A equação não é de segundo grau.")
else:
    # Calcula o discriminante
    delta = b**2 - 4*a*c
    
    if delta > 0:
        # Duas raízes reais distintas
        x1 = (-b + math.sqrt(delta)) / (2*a)
        x2 = (-b - math.sqrt(delta)) / (2*a)
        print(f"As raízes são: x1 = {x1:.2f} e x2 = {x2:.2f}")
    elif delta == 0:
        # Uma raiz real (raiz dupla)
        x = -b / (2*a)
        print(f"A raiz dupla é: x = {x:.2f}")
    else:
        # Duas raízes complexas
        real_part = -b / (2*a)
        imaginary_part = math.sqrt(abs(delta)) / (2*a)
        print(f"As raízes complexas são: x1 = {real_part:.2f} + {imaginary_part:.2f}i e x2 = {real_part:.2f} - {imaginary_part:.2f}i")
